themes/digital.gov/src/js/glossary.js

```diff
@@ -0,0 +1,251 @@
+    <svg id="${id}-icon" class="${values.iconClass} usa-icon dg-icon dg-icon--large" aria-hidden="true" focusable="false" role="img" hidden>
+      <use class="${values.iconClass}"xlink:href="${glossaryIcon}"></use>
+    </svg>
+    <div id="${id}" class="${values.definitionClass}" hidden>${values.definition}</div>
+  </li>`;
+  return template;
+}
+
+/**
+ * Generate list markup and insert into list element to display
+ * @param {object} terms json data list of terms and definitions
+ */
+
+function generateListMarkup(terms) {
+  let termsMarkup = "";
+  terms.forEach((term, i) => {
+    const values = {
+      term: term.term,
+      definition: term.definition,
+      definitionClass: cssClasses.definition,
+      itemClass: cssClasses.item,
+      termClass: cssClasses.term,
+      iconClass: cssClasses.icon,
+      termId: i,
+    };
+    termsMarkup += itemTemplate(values);
+  });
+  listElement.insertAdjacentHTML("beforeend", termsMarkup);
+}
+
+/**
+ * Construct new list object from list.js library
+ */
+
+function initializeList() {
+  const listClass = selectors.list.slice(1);
+  const searchClass = selectors.search.slice(1);
+  const options = {
+    valueNames: [cssClasses.term],
+    listClass,
+    searchClass,
+  };
+  list = new List(glossaryBody, options);
+  definitions = glossaryBody.querySelectorAll(`.${cssClasses.term}`);
+}
+
+/**
+ * Hide glossary and disable all focusable elements
+ */
+
+function hideGlossary() {
+  if (toggleButtonTrigger) {
+    glossaryBody.setAttribute("aria-hidden", "true");
+    closeButton.setAttribute("disabled", "");
+    search.setAttribute("disabled", "");
+    definitions.forEach((definition) =>
+      definition.setAttribute("disabled", "")
+    );
+    toggleButtonTrigger.focus({ preventScroll: true });
+    toggleButtonTrigger = null;
+  }
+}
+
+/**
+ * Show glossary and enable all focusable elements
+ * @param {element} toggleButton button which was clicked to open the glossary
+ */
+
+function showGlossary(toggleButton) {
+  if (!toggleButtonTrigger) {
+    glossaryBody.setAttribute("aria-hidden", "false");
+    closeButton.removeAttribute("disabled");
+    search.removeAttribute("disabled");
+    search.focus();
+    definitions.forEach((definition) => definition.removeAttribute("disabled"));
+    toggleButtonTrigger = toggleButton;
+  } else {
+    hideGlossary();
+  }
+}
+
+/**
+ * Toggle glossary term expansion when clicked
+ * @param {event} event fired when term is clicked
+ */
+
+function handleTermClick(e) {
+  if (e.target.matches(`.${cssClasses.term}`)) {
+    const termNumber = e.target.getAttribute("aria-controls");
+    e.target.setAttribute(
+      "aria-expanded",
+      e.target.getAttribute("aria-expanded") === "true" ? "false" : "true"
+    );
+    glossaryBody
+      .querySelectorAll(`#${termNumber}, #${termNumber}-icon`)
+      .forEach((definition) => definition.toggleAttribute("hidden"));
```

Comment:
It's generally not recommended to style based on the `hidden` attribute, unless you're using something like `display: none`.

We should instead use an `active` class that will change from the default state.